aocd/examples.py

```diff
@@ -55,16 +55,19 @@ def from_raw(cls, html):
         articles = soup.find_all("article")
         if len(articles) == 0:
             raise ExampleParserError("no <article> found in html")
-        elif len(articles) == 1:
-            [article_a] = articles
-            a_raw = str(article_a)
-            article_b = b_raw = None
-        elif len(articles) == 2:
-            article_a, article_b = articles
-            a_raw = str(article_a)
-            b_raw = str(article_b)
-        else:
+        if len(articles) > 2:
             raise ExampleParserError("too many <article> found in html")
+
+        article_a = articles.pop(0)
+        assert isinstance(article_a, bs4.Tag)
+        a_raw = str(article_a)
+
+        article_b = b_raw = None
+        if articles:
+            article_b = articles.pop(0)
+            assert isinstance(article_b, bs4.Tag)
+            b_raw = str(article_b)
+
```

Comment:
The type checker was quite unhappy with the original approach here. I believe this rewrite should have identical behavior from an end-user perspective.